Sources/AppKitNavigationShim/include/shim.h

```diff
@@ -0,0 +1,19 @@
+#if __has_include(<TargetConditionals.h>)
+#include <TargetConditionals.h>
+
+#if __has_include(<AppKit/AppKit.h>) && !TARGET_OS_MACCATALYST
+@import AppKit;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NSViewController (AppKitNavigation)
+
+@property BOOL hasViewAppeared;
+@property (nullable) void (^ onDismiss)();
+@property NSArray<void (^)()> *onViewAppear;
```

Comment:
We've had to prefix these in UIKitNavigation to avoid collisions with external code.

```suggestion
@property BOOL _AppKitNavigation_hasViewAppeared;
@property (nullable) void (^ _AppKitNavigation_onDismiss)();
@property NSArray<void (^)()> *_AppKitNavigation_onViewAppear;
```